e2e-tests/insurance/cypress/e2e/journeys/export-contract/change-your-answers/agent-service/change-your-answers-agent-service-description.spec.js

```diff
@@ -0,0 +1,84 @@
+import { field, summaryList } from '../../../../../../../pages/shared';
+import { EXPECTED_SINGLE_LINE_STRING } from '../../../../../../../constants';
+import { INSURANCE_ROUTES } from '../../../../../../../constants/routes/insurance';
+import FIELD_IDS from '../../../../../../../constants/field-ids/insurance/export-contract';
+import application from '../../../../../../../fixtures/application';
+
+const {
+  ROOT,
+  EXPORT_CONTRACT: {
+    CHECK_YOUR_ANSWERS,
+    AGENT_SERVICE_CHANGE,
+  },
+} = INSURANCE_ROUTES;
+
+const {
+  AGENT_SERVICE: { SERVICE_DESCRIPTION },
+} = FIELD_IDS;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Export contract - Change your answers - Agent service - description - As an Exporter, I want to be able to review my input regarding whether an agent helped me win the export contract, So that I can be assured I am providing UKEF with the right information', () => {
+  let referenceNumber;
+  let checkYourAnswersUrl;
+
+  before(() => {
+    cy.completeSignInAndGoToApplication({}).then(({ referenceNumber: refNumber }) => {
+      referenceNumber = refNumber;
+
+      cy.completeExportContractSection({
+        isUsingAgent: true,
+      });
+
+      checkYourAnswersUrl = `${baseUrl}${ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`;
+
+      cy.assertUrl(checkYourAnswersUrl);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  after(() => {
+    cy.deleteApplication(referenceNumber);
+  });
+
+  describe(SERVICE_DESCRIPTION, () => {
+    const fieldId = SERVICE_DESCRIPTION;
+
+    describe('when clicking the `change` link', () => {
+      it(`should redirect to ${AGENT_SERVICE_CHANGE}`, () => {
+        cy.navigateToUrl(checkYourAnswersUrl);
+
+        summaryList.field(fieldId).changeLink().click();
+
+        cy.assertChangeAnswersPageUrl({ referenceNumber, route: AGENT_SERVICE_CHANGE, fieldId });
+      });
+    });
+
+    describe('form submission with a new answer', () => {
+      const newAnswer = `${application.EXPORT_CONTRACT.AGENT_SERVICE[fieldId]} additional text`;
+
+      beforeEach(() => {
+        cy.navigateToUrl(checkYourAnswersUrl);
+
+        summaryList.field(fieldId).changeLink().click();
+
+        cy.keyboardInput(field(fieldId).textarea(), newAnswer);
```

Comment:
is there a change your answers cy function to do this? and check that the correct input is displayed?

Reply:
We do have a `changeAnswerField`, but this won't work for a textarea. We could create `changeAnswerTextareaField` and rename the existing one to `changeAnswerTextInputField`, but I think this should be done across all tests in one go.

I've updated the assertion to use `checkSummaryList`, with a new, optional `newAnswer` param for the textarea.